internal/core/core.go

```diff
@@ -0,0 +1,254 @@
+package core
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"regexp"
+	"sync"
+	"text/template"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/binding"
+	"github.com/openconfig/ondatra/eventlis"
+	"google.golang.org/grpc"
+
+	fpb "github.com/openconfig/gnoi/file"
+	opb "github.com/openconfig/ondatra/proto"
+)
+
+var (
+	vendorCoreFilePath = map[opb.Device_Vendor]string{
+		opb.Device_JUNIPER: "/var/core/",
+		opb.Device_CISCO:   "/misc/disk1/",
+		opb.Device_NOKIA:   "/var/core/",
+		opb.Device_ARISTA:  "/var/core/",
+	}
+	vendorCoreFileNamePattern = map[opb.Device_Vendor]*regexp.Regexp{
+		opb.Device_JUNIPER: regexp.MustCompile(".*.tar.gz"),
+		opb.Device_CISCO:   regexp.MustCompile("/misc/disk1/.*core.*"),
+		opb.Device_NOKIA:   regexp.MustCompile("/var/core/coredump-.*"),
+		opb.Device_ARISTA:  regexp.MustCompile("/var/core/core.*"),
+	}
+)
+
+var (
+	Validator = validator{
+		duts: map[string]*checker{},
+	}
+)
+
+type FileInfo struct {
+	Name     string
+	Path     string
+	Modified uint64
+	Existing bool
+}
+
+type CoreFiles map[string]FileInfo
+
+type CoreFileEntry struct {
+	DUT            string
+	CollectionTime time.Time
+	Files          CoreFiles
+}
+
+type checker struct {
+	dut        binding.DUT
+	fileClient fpb.FileClient
+
+	mu        sync.Mutex
+	snapshots []CoreFileEntry
+	startTime time.Time
+	endTime   time.Time
+	delta     CoreFiles
+}
+
+func newChecker(dut binding.DUT) (*checker, error) {
+	c := &checker{
+		dut:       dut,
+		delta:     map[string]FileInfo{},
+		startTime: time.Now(),
+	}
+	gClients, err := c.dut.DialGNOI(context.Background(), grpc.WithBlock())
+	if err != nil {
+		return nil, err
+	}
+	c.fileClient = gClients.File()
+	return c, nil
+}
+
+func (c *checker) Check() error {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	cores, err := c.checkCores()
+	if err != nil {
+		return err
+	}
+	c.snapshots = append(c.snapshots, cores)
```

Comment:
It looks like the content of this snapshots slice is never looked at, only whether this has been called once? Could this be more simply tracked with a boolean?

Reply:
i wanted to use snapshots to be able to use the timestamps to actually determine when a core was found then compare to what test was running